onnxruntime/core/providers/cuda/tensor/trilu.cc

```diff
@@ -34,12 +34,12 @@ Status Trilu::ComputeInternal(OpKernelContext* ctx) const {
   const Tensor& input = *input_ptr;
   const auto& shape = input.Shape();
   const auto& input_dims = shape.GetDims();
-  int32_t rank = gsl::narrow_cast<int32_t>(input_dims.size());
+  auto rank = gsl::narrow_cast<int32_t>(input_dims.size());
   if (rank < 2) {
     return Status(ONNXRUNTIME, INVALID_ARGUMENT, "Input tensor should have a rank of at least 2");
   }
   Tensor* output = ctx->Output(0, shape);
-  int64_t matrix_size = input_dims[rank - 1] * input_dims[rank - 2];
+  auto matrix_size = input_dims[static_cast<int64_t>(rank) - 1] * input_dims[static_cast<int64_t>(rank) - 2];
```

Comment:
can you remove the narrow cast for rank on line 37 instead of this?

Reply:
I think at the time we did the narrow cast to reduce the computation overhead.